### DISCUSSION TITLE
Suppress warnings for MDXProvider

### DISCUSSION THREAD

**Discussion by deadcoder0904** [mdx-js/mdx#1296] [Q&A] [answered]
So I had this issue of [Converting MDX to HTML](https://github.com/mdx-js/mdx/discussions/1295) which I solved but it gives warnings in the console like:

```
> node ./.next/server/scripts/build-rss.js

Warning: Invalid DOM property `frameborder`. Did you mean `frameBorder`?
    in iframe
    in MDXCreateElement
    in div
    in MDXCreateElement
    in Fragment
    in wrapper
    in MDXCreateElement
    in MDXContent
    in Context.Provider
    in MDXProvider
Warning: Invalid DOM property `allowfullscreen`. Did you mean `allowFullScreen`?
    in iframe
    in MDXCreateElement
    in div
    in MDXCreateElement
    in Fragment
    in wrapper
    in MDXCreateElement
    in MDXContent
    in Context.Provider
    in MDXProvider
Warning: Invalid DOM property `class`. Did you mean `className`?
    in figcaption
    in MDXCreateElement
    in figure
    in MDXCreateElement
    in Fragment
    in wrapper
    in MDXCreateElement
    in MDXContent
    in Context.Provider
    in MDXProvider
```

I have these `package.json` scripts:

```json
"scripts": {
    "dev": "next dev",
    "build": "next build && next export && npm run build:rss",
    "start": "next start",
    "build:rss": "node ./.next/server/scripts/build-rss.js"
  },
```

And I am directly running `npm run build` & it gives those warnings.

Can I suppress these warnings?

**Comment by wooorm** [comment, marked as answer]
Why suppress them? They’re good warnings, that point to something that should be fixed?

**Reply by deadcoder0904** [reply]
@wooorm I'm using it to generate RSS feed only once so doesn't make sense to have them because otherwise, I can't generate RSS feed while using `Next + MDX` setup without using `MDXProvider`.

The warning should be applicable in JSX context, not HTML/XML context I think.

**Reply by wooorm** [reply]
[XY problem](https://meta.stackexchange.com/questions/66377/what-is-the-xy-problem), I believe.

You’re passing HTML to MDX. MDX is Markdown - HTML + JSX. We don’t handle HTML.
I’m not sure why you are passing HTML to MDX. Maybe you don’t have to do that.
I don’t think there’s any Markdown involved?
If you have to do it, you’ll need something to turn HTML into JSX. I believe there are projects that do that. I’ve done it with rehype-parse and then using rehype-react.

**Reply by deadcoder0904** [reply]
Not sure if it's an XY problem tbh. Maybe my wording is incorrect. I am not passing HTML to MDX.

I am creating a blog using Next.js & MDX. My RSS script needs HTML to output `feed.xml`.

Now as every post is written in MDX, I have to use `MDXProvider` & `ReactDOMServer.renderToStaticMarkup` (as you can see https://github.com/mdx-js/mdx/discussions/1295#discussioncomment-100333) to convert the MDX thing into HTML so I can actually generate RSS feed.

Due to this, it gives a warning which I think is unnecessary because every time I generate RSS, it will show this warning which can be disabled by passing a flag like `disableWarning` like `<MDXProvider disableWarning></MDXProvider>`.

What do you think?

**Reply by wooorm** [reply]
But where is the `class` on `figcaption` coming from then? Because class is html, className is JSX

**Reply by deadcoder0904** [reply]
I searched in the repo & one of the [index.mdx](https://github.com/tailwindlabs/blog.tailwindcss.com/blob/master/src/pages/simon-vrachliotis-joins-tailwind-labs/index.mdx) contains `figcaption` using `class`:

```js
<figcaption class="italic">
    We know this video isn't from Laracon, but this recording turned out better :)
</figcaption>
```

So when I wrap this MDX into `MDXProvider`, it throws the warning as JSX uses `className` instead of `class`.

**Reply by wooorm** [reply]
So these are warnings you can fix? Same w/ frameborder and allowfullscreen!

**Reply by deadcoder0904** [reply]
I changed that `index.mdx` file to use `className` & it did work. Although, I was looking for a way to suppress the warnings. I guess it's not possible currently :)

**Reply by ChristianMurphy** [reply]
@deadcoder0904 MDX uses JSX not HTML, the warning you are seeing is a JSX warning thrown by React itself, not MDX.

**Reply by deadcoder0904** [reply]
Yep, just understood it. It was a mistake in MDX files itself by the author 😂

Sorry for the trouble & your time 🙏
